### DISCUSSION TITLE
Setting background on qtile

### DISCUSSION THREAD

**Discussion by AlphabetsAlphabets** [qtile/qtile#2445] [General]
In my qtile config I have this line:
`lazy.spawn("nitrogen --restore")`  

But it doesn't set the background automatically. Is there a reason why?

**Comment by m-col** [comment]
The `lazy` interface is used for doing something later on, for example, binding `lazy.spawn("nitrogen --restore")` to a key to call `nitrogen` later when the key is pressed. If you want to call `nitrogen` when the config is read you could use `qtile.cmd_spawn("nitrogen --restore")` or something from the `subprocess` module like `subprocess.Popen`.

**Reply by AlphabetsAlphabets** [reply]
I took a look at it, and is [this](http://docs.qtile.org/en/latest/manual/ref/commands.html?highlight=cmd%20#libqtile.core.manager.Qtile.cmd_spawn) the same `cmd-spawn` you were referring to? I found another `qtile` this time from `libqtile.qtile` and that doesn't do anything either.

I also found [this](http://docs.qtile.org/en/latest/manual/config/hooks.html?highlight=spawn#autostart) and [this](http://docs.qtile.org/en/latest/manual/ref/commands.html?highlight=cmd%20#libqtile.core.manager.Qtile.cmd_spawn)

**Comment by Fabien-Bernard** [comment]
When I used nitrogen, I added "nitrogen --restore &" to my .xinitrc, not to my qtile config.

Now I use the native setting of qtile to manage my wallpaper, so I don't need nitrogen anymore.
See also the wiki for options: https://github.com/qtile/qtile/wiki/wallpapers

**Reply by AlphabetsAlphabets** [reply]
This worked now. That was odd, previously it didn't work when I used that setting. But I'm glad it did now. Thank you.
